routing/routing_helpers.cpp

```diff
@@ -35,5 +42,34 @@ void ReconstructRoute(IDirectionsEngine * engine, IRoadGraph const & graph,
   route.SetTurnInstructions(move(turnsDir));
   route.SetStreetNames(move(streetNames));
   route.SetAltitudes(move(altitudes));
+
+  vector<traffic::SpeedGroup> traffic;
+  traffic.reserve(routeSegs.size());
+  if (trafficColoring)
+  {
+    for (TrafficInfo::RoadSegmentId const & rp : routeSegs)
+    {
+      auto const it = trafficColoring->find(rp);
+      SpeedGroup segTraffic =  (it == trafficColoring->cend()) ? SpeedGroup::Unknown
+                                                               : it->second;
+      // @TODO It's written to compensate an error. The problem is in case of any routing except for osrm
+      // all route points except for begining and ending are duplicated.
+      // See a TODO in BicycleDirectionsEngine::Generate() for details.
+      // At this moment the route looks like:
+      // 0----1    4----5
+      //      2----3    6---7
+      // This route composes of 4 edges and there're 4 items in routeSegs.
+      // But it has 8 items in routeGeometry.
+      // So for segments 0-1 and 1-2 let's set routeSegs[0]
+      // for segments 2-3 and 3-4 let's set routeSegs[1]
+      // for segments 4-5 and 5-7 let's set routeSegs[2]
+      // and for segment 6-7 let's set routeSegs[3]
+      traffic.insert(traffic.end(), {segTraffic, segTraffic});
```

Comment:
Here I'd rather use insert because to has interval/initializer_list signature. Do you think using push_back twice is better here?